web/src/modules/layout/components/HeaderActionItemDropDown.js

```diff
@@ -64,63 +62,45 @@ export const DropDownContainer = styled.div`
 `
 
 type PropsType = {|
-  children: React.Element<*>,
+  children: (closeDropDown: () => void) => React.Element<*>,
   theme: ThemeType,
   iconSrc: string,
   text: string
 |}
 
-type StateType = {|
-  dropDownActive: boolean
-|}
-
 /**
  * Designed to work as an item of a HeaderActionBar. Once clicked, the child node becomes visible right underneath the
  * Header. Once the user clicks outside, the node is hidden again. Additionally, the inner node gets a
  * closeDropDownCallback through its props to close the dropDown and hide itself.
  */
-export class HeaderActionItemDropDown extends React.Component<PropsType, StateType> {
-  componentDidUpdate () {
-    /* https://www.npmjs.com/package/react-tooltip#1-using-tooltip-within-the-modal-eg-react-modal- */
-    ReactTooltip.rebuild()
-  }
+const HeaderActionItemDropDown = (props: PropsType) => {
```

Comment:
So do you think we should gradually be moving to hooks and functions?

Reply:
Yes we should. Our codebase already feels legacy :grimacing: